include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1769,16 +1769,12 @@ __pattern_hetero_set_op(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _F
                         _ForwardIterator1 __last1, _ForwardIterator2 __first2, _ForwardIterator2 __last2,
                         _OutputIterator __result, _Compare __comp, _SetTag __set_tag)
 {
-    typedef typename std::iterator_traits<_ForwardIterator1>::difference_type _Size1;
+    using _SizeType = std::common_type_t<typename std::iterator_traits<_ForwardIterator1>::difference_type,
+                                         typename std::iterator_traits<_ForwardIterator2>::difference_type>;
 
-    const _Size1 __n1 = std::distance(__first1, __last1);
-    _Size1 __output_size = __n1;
-    if constexpr (_SetTag::__can_write_from_rng2_v)
-    {
-        const _Size1 __n2 = std::distance(__first2, __last2);
-        // one shot algorithm can write from set 1 or set 2, whereas old algorithm can only write from set 1.
-        __output_size = __n1 + __n2;
-    }
+    const _SizeType __n1 = std::distance(__first1, __last1);
```

Comment:
In another cases @akukanov prefer `__last1 - __first1` of course if they are random-access iterators...